File: src/main/java/com/demonwav/statcraft/magic/EntityCode.java
/*
 * StatCraft Bukkit Plugin
 *
 * Copyright (c) 2016 Kyle Wood (DemonWav)
 * http://demonwav.com
 *
 * MIT License
 */

package com.demonwav.statcraft.magic;

import org.apache.commons.lang.WordUtils;
import org.bukkit.entity.Entity;
import org.bukkit.entity.Skeleton;

public enum EntityCode {

    OTHER,
    SKELETON,
    WITHER_SKELETON;

    public static EntityCode fromEntity(Entity entity) {
        if (entity instanceof Skeleton) {
            Skeleton skeleton = (Skeleton) entity;
            if (skeleton.getSkeletonType() == Skeleton.SkeletonType.NORMAL)
                return EntityCode.SKELETON;
            else if (skeleton.getSkeletonType() == Skeleton.SkeletonType.WITHER)
                return EntityCode.WITHER_SKELETON;
        }
        return EntityCode.OTHER;
    }

    public String getName(String name) {
        switch (this) {
            case OTHER:
                return name;
            case SKELETON:
            case WITHER_SKELETON:
                return WordUtils.capitalizeFully(this.name().replace('_', ' '));
            default:
                return "";
        }
    }
}

File: src/main/kotlin/com/demonwav/statcraft/magic/EntityCode.kt
/*
 * StatCraft Bukkit Plugin
 *
 * Copyright (c) 2016 Kyle Wood (DemonWav)
 * https://www.demonwav.com
 *
 * MIT License
 */

package com.demonwav.statcraft.magic

import org.apache.commons.lang.WordUtils
import org.bukkit.entity.Entity
import org.bukkit.entity.Skeleton

enum class EntityCode {

    OTHER,
    SKELETON,
    WITHER_SKELETON;

    fun getName(name: String): String {
        when (this) {
            OTHER -> return name
            SKELETON, WITHER_SKELETON -> return WordUtils.capitalizeFully(this.name.replace('_', ' '))
        }
    }

    companion object {
        @JvmStatic
        fun fromEntity(entity: Entity): EntityCode {
            if (entity is Skeleton) {
                if (entity.skeletonType == Skeleton.SkeletonType.NORMAL)
                    return EntityCode.SKELETON
                else if (entity.skeletonType == Skeleton.SkeletonType.WITHER)
                    return EntityCode.WITHER_SKELETON
            }
            return EntityCode.OTHER
        }
    }
}

